File: angular/lti/Doctor_Angular/src/app/model/doctor.ts
import { Patient } from "./patient";

export class Doctor {
 id:number=0;
 name:string='';
 age:number=0;

 //patientSet:Patient[]=[]
 patientSet:Array<{id:number,
    name:string,
    age:number,
    location:string;
}>=[];
 

}

File: kochi 2025_2024_dec/relationship/spring boot/lti onetomany_uni/Doctor_Spring/src/main/java/org/example/entity/Doctor.java
package org.example.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.util.Objects;
import java.util.Set;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString(exclude = {"patinetSet"})
@EqualsAndHashCode(exclude ={"patientSet"} )

public class Doctor {

    @Id
    @GeneratedValue
    private int id;
    private String name;
    private int age;

    @OneToMany(cascade = CascadeType.ALL,fetch = FetchType.EAGER)
    @JsonIgnoreProperties("doctor")
    private Set<Patient> patientSet;

}
